src/vm/methodtablebuilder.cpp

```diff
@@ -9744,6 +9744,17 @@ void MethodTableBuilder::CheckForSystemTypes()
 
             pMT->SetComponentSize(2);
         }
+        else if (strcmp(name, g_Utf8StringName) == 0 && strcmp(nameSpace, g_SystemNS) == 0)
+        {
+            // Utf8Strings are not "normal" objects, so we need to mess with their method table a bit
+            // so that the GC can figure out how big each string is...
+            DWORD baseSize = ObjSizeOf(Utf8StringObject) + sizeof(BYTE);
+            pMT->SetBaseSize(baseSize); // NULL character included
+
+            GetHalfBakedClass()->SetBaseSizePadding(baseSize - bmtFP->NumInstanceFieldBytes);
```

Comment:
`GetHalfBakedClass` is this a real method 🤣

Reply:
<del>method</del> function (it's c++) 😉